src/logger.coffee

```diff
@@ -5,114 +5,237 @@ es = require 'event-stream'
 Lock = require 'rwlock'
 { checkTruthy } = require './lib/validation'
 
-LOG_PUBLISH_INTERVAL = 110
+class NativeLoggerBackend
+	constructor: ->
+		@logPublishInterval = 1000
+		@maxLogLinesPerBatch = 10
+		@maxQueueSize = 100
+		@maxLineLength = 300
+		@publishQueue = []
+		@intervalHandle = null
+		@publishEnabled = false
+		@offlineMode = false
+		@shouldStop = false
+
+	publishLoop: =>
+		startTime = process.hrtime()
+		Promise.try =>
+			return if @offlineMode or !@publishEnabled or @publishQueue.length is 0
+			currentBatch = _.slice(@publishQueue, 0, @maxLogLinesPerBatch)
+			@publishQueue = _.slice(@publishQueue, @maxLogLinesPerBatch)
+			# Silently ignore errors sending logs (just like with pubnub)
+			@apiBinder.logBatch(currentBatch)
+		.catchReturn()
+		.then =>
+			elapsedTime = process.hrtime(startTime)
+			elapsedTimeMs = elapsedTime[0] * 1000 + elapsedTime[1] / 1e6
+			nextDelay = Math.min(@logPublishInterval - elapsedTimeMs, 0)
+			Promise.delay(nextDelay)
+		.then =>
+			if @shouldStop
```

Comment:
NIT: Seems like `if !@shouldStop @publishLoop()` would be more straightforward

Reply:
Done